pkg/apis/infinispan/v1/types_util.go

```diff
@@ -135,6 +138,18 @@ func (ispn *Infinispan) ApplyDefaults() {
 	}
 }
 
+func (ispn *Infinispan) ApplyMonitoringAnnotation(supported bool) {
+	if supported {
```

Comment:
This seems backwards to me. If you're providing the `boolean` flag as a parameter, and no other action is taken if this flag is false, then it makes more sense for the call the `ApplyMonitoringAnnotation` to be surrounded by an `if` statement where it's called.

For example in `infinispan_controller.go`

```go
if isTypeSupported(consts.ServiceMonitorType) {
    infinispan.ApplyMonitoringAnnotation()
}
```

Reply:
Yes, make sense, will fix